### ISSUE TITLE
Logic Engine call leads to immediate taskmanager segfault exit

### ISSUE THREAD

**Issue by StevenCTimm** [HEPCloud/decisionengine#240]
We have observed in two recent builds against trunk, the following behavior:

4 decision engine channels go to ERROR state and their task managers exit,  with the last thing in their log being:

1:03
2020-12-17 10:16:18,102 - root - LogicEngine - 24669 - MainThread - INFO - LE: calling evaluate_facts
2020-12-17 10:16:18,102 - root - LogicEngine - 24669 - MainThread - INFO - Evaluated Fact: allow_aws_config -> Value: True -> TypeOf(Value): <class 'bool'>
2020-12-17 10:16:18,102 - root - LogicEngine - 24669 - MainThread - INFO - LE: calling execute

Evidently the name of the c++ logic engine library was recently changed along with the manner in which it was built.

Early analysis by Pat Riehecky indicates there is a segfault showing in dmesg at the point of failure.

**Comment by StevenCTimm** [comment]
[92110.274302] python3[28325]: segfault at 7ffd404d0fc0 ip 00007f00858520b1 sp 00007ffd404d0fc0 error 6 in libc-2.17.so[7f00857d0000+1c4000]

This is the segfault message.

Current version of libc on fermicloud117 is 

glibc-2.17-317.el7.x86_64

**Comment by StevenCTimm** [comment]
Pat has got it to dump a core, now we need someone to do gdb.

**Comment by DmitryLitvintsev** [comment]
git bisect exercise yields this patch as a culprit:

```
8815dc6d7233fc3695b7d9bb3e33f142fb5ebeca Logic-engine cleanups (#211)
```

**Comment by StevenCTimm** [comment]
Kyle when do we expect a patched version of this to be available to test?

**Comment by knoepfel** [comment]
A patch is available at PR #245.  Awaiting review from decision-engine developers.

**Comment by StevenCTimm** [comment]
Is there a build with the above patch? Has the patch been merged to trunk?

**Comment by StevenCTimm** [comment]
I have installed a version of framework built from trunk which incorporates Kyle's PR above and can confirm that the logic engine starts correctly and functions normally.

**Comment by StevenCTimm** [comment]
This was resolved in version 1.5.  Closing.
